apps/system/js/rocketbar.js

```diff
@@ -567,6 +568,14 @@
       if (Service.currentApp.isPrivateBrowser()) {
         this.setInput('');
       }
+
+      // Only record the search if we have internet connectivity
+      if(navigator.onLine) {
+        var provider = window.SearchProvider('title');
```

Comment:
It returns the user's currently selected search provider: (e.g. 'google' 'yahoo')  If there is some error condition, it returns a false. I could check the return and not send the event.  I think we should probably log an error in this case because I don't think it is possible to have a setup where there is no selected search provider.

So, I'm glad you asked this because I did some more investigation on this and it looks like this function does not work after I do a |make reset-gaia|.  It seems to only work after I reboot the device after the make-reset gaia.  I'm going to look some more into this.